frontend/tests/integration/components/mapbox/filter-features-test.js

```diff
@@ -0,0 +1,70 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import { registerEventHandler } from '../../../helpers/mapbox/mapbox-stub-helpers';
+
+const DEFAULT_MAPBOX_GL_INSTANCE = {
+  addSource: () => {},
+  addLayer: () => {},
+  removeLayer: () => {},
+  removeSource: () => {},
+  getStyle: () => ({}),
+  queryRenderedFeatures: () => [],
+  on: () => {},
+  off: () => {},
+};
+
+module('Integration | Component | mapbox/filter-features', function(hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(async function() {
+    this.events = {};
+    this.layers = [];
+    this.hoveredFeature = null;
+    this.filters = {}
+    this.map = {
+      ...DEFAULT_MAPBOX_GL_INSTANCE,
+      addLayer: ({ id }) => {
+        this.layers.push(id);
+      },
+      queryRenderedFeatures: () => {
+        return [{
+          type: 'Feature',
+          properties: {
+            geoid: 'test',
+          },
+        }]
+      },
+      setFilter: (layerId, filter) => {
+        this.filters[layerId] = filter;
+      },
+      // On render, component templates like `current-map-position.js`
+      // will associate action handlers to fired mouse events and 
+      // mantain a list of these associations. 
+      // We simulate this association list locally (with the `events` object)
+      // so that we can make ad hoc calls to action handlers. See comment below.
+      on: (event, action) => {
+        registerEventHandler(this.events, event, action);
+      }
+    };
+
+  });
+
+  test('it renders', async function(assert) {
+
+    this.idList = ['1', '12', '123'];
+
+    await render(hbs`
+      {{mapbox/filter-features
+        map=(hash instance=this.map)
+        layerId='test'
+        filterById='id'
+        featureIds=this.idList
+      }}
+    `);
+
+    assert.equal(JSON.stringify(this.filters['test']), JSON.stringify(['in', 'id', '1', '12', '123']));
+  });
```

Comment:
Can you describe what this asserts in `test('it renders`?

I also don't get why we use all those JSON methods in the assertion... I think it will stringify the data anyways. Otherwise, you could use `assert.deepEqual` if you want to assert equality of non-primitive data structures.

Reply:
Oh sweet, deepEqual is perfect 🙇 🙏  Thank you
Yah I was trying to get around it thinking the two arrays are different despite an equal serialization.